File: Python_Workspace/BaseballGame.py
import random
rn = ["0", "0", "0"] 
s_cnt = 0 # 스트라이크 갯수
b_cnt = 0 # 볼 갯수 

def reset():
    global rn
    global s_cnt
    global b_cnt
    
    rn[0] = str(random.randint(0, 9))
    rn[1] = rn[0]
    rn[2] = rn[0]
    while (rn[0] == rn[1]):
        rn[1] = str(random.randint(0, 9))
    while (rn[0] == rn[2] or rn[1] == rn[2]):
        rn[2] = str(random.randint(0, 9))

    s_cnt = 0
    b_cnt = 0

def check(num):
    global rn
    global s_cnt
    global b_cnt

    for i in range(0, 3): 
        for j in range(0, 3): 
            if(num[i] == str(rn[j]) and i == j): s_cnt += 1 
            elif(num[i] == str(rn[j]) and i != j): b_cnt += 1 

    if(s_cnt != 3):
        print(str(s_cnt) + " strike, " + str(b_cnt) + " ball." + '\n')
        print("============================")
    return s_cnt

def main() :
    global s_cnt
    global b_cnt
    cnt = 1
    print("S T A R T")
    print("============================"+'\n')
    while(cnt == 1):
        reset()
        while (s_cnt < 3): 
            num = str(input("Guess numbers (000 ~ 999): "+'\n')) 
            s_cnt = 0
            b_cnt = 0
            s_cnt = check(num)
            if(s_cnt == 3): break

        print("YES! The secret number is " + str(num) + "! You have won!" + '\n')
        recycle = str(input("Do you want to play again? (y/n)"))
        if(recycle == "y" or recycle == "Y"): cnt = 1
        else : cnt = 0

    print("Game Finish~ Bye~")
if __name__ == "__main__":
    main()

File: C_Workspace/BaseballGame.cpp
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

int rn[4],s_cnt,b_cnt;
int num[4];
void reset(){
	srand(time(NULL));
	rn[0] = rand() % 10;
	rn[1] = rn[0];
	rn[2] = rn[0];
    while (rn[0] == rn[1]) rn[1] = rand() % 10;
    while (rn[0] == rn[2] || rn[1] == rn[2]) rn[2] = rand() % 10;

    s_cnt = 0;
    b_cnt = 0;
}

int check(int ball){
	num[0] = ball / 100;
	ball = ball % 100;
	num[1] = ball / 10;
	ball = ball % 10;
	num[2] = ball;

	for(int i=0;i<3;i++){
		for(int j=0;j<3;j++){
			if(num[i] == rn[j] && i == j) s_cnt += 1;
			else if (num[i] == rn[j] && i != j) b_cnt += 1;
		}
	}
	if(s_cnt != 3){
		if(s_cnt == 0 && b_cnt == 0) printf("out!\n");
		else printf("%d strike, %d ball.\n",s_cnt,b_cnt);
        printf("============================\n");
	}
    return s_cnt;
}

int main(){
	int cnt = 1,n;
	char recycle;
    printf("============================\n");
    printf("S T A R T\n");
    while(cnt == 1){
        reset();
		while (s_cnt < 3){
			printf("Guess numbers (000 ~ 999): ");
			scanf("%d",&n);
            s_cnt = 0;
            b_cnt = 0;
            s_cnt = check(n);
            if(s_cnt == 3) break;
		}

		printf("Correct, answer is %d%d%d\n",num[0],num[1],num[2]);
		printf("Play again? <y or n>: ");
		scanf("%s",&recycle);
        if(recycle == 'y' || recycle == 'Y') cnt = 1;
        else cnt = 0;
	}
    printf("Game Finish~ Bye~\n");
	return 0;	
}